x-pack/legacy/plugins/siem/public/pages/detection_engine/components/signals_histogram_panel/signals_histogram/helpers.tsx

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { HistogramData, SignalsAggregation, SignalsBucket, SignalsGroupBucket } from '../types';
+import { SignalSearchResponse } from '../../../../../containers/detection_engine/signals/types';
+import * as i18n from '../translations';
+
+export const formatSignalsData = (
+  signalsData: SignalSearchResponse<{}, SignalsAggregation> | null
+) => {
+  const groupBuckets: SignalsGroupBucket[] =
+    signalsData?.aggregations?.signalsByGrouping?.buckets ?? [];
+  let result: HistogramData[] = [];
+  groupBuckets.forEach(({ key: group, signals }) => {
+    const signalsBucket: SignalsBucket[] = signals?.buckets ?? [];
+
+    result = [
+      ...result,
+      ...signalsBucket.map(({ key, doc_count }: SignalsBucket) => ({
+        x: key,
+        y: doc_count,
+        g: group,
+      })),
+    ];
+  });
+
+  return result;
+};
+
+export const getSignalsHistogramQuery = (
+  stackByField: string,
+  from: number,
+  to: number,
+  additionalFilters: object[]
```

Comment:
Is there type here same as `ESQuery`?

Reply:
So it would be ideal to use `esFilters.Filter[]`, but it differs from the return value of `esQuery.buildEsQuery()` which is the what we use to build the additional filters. To help narrow types though I've updated `additionalFilters` to match what `esQuery.buildEsQuery()` returns. Thanks for pointing this out!

i.e.
``` ts
additionalFilters: Array<{ bool: { filter: unknown[]; should: unknown[]; must_not: unknown[]; must: unknown[] }; }>
```